homework20/PetrolStation.java

```diff
@@ -0,0 +1,47 @@
+package homework20;
+
+import java.util.concurrent.Semaphore;
+
+public class PetrolStation {
+
+    private int amount;
+    private final Semaphore semaphore = new Semaphore(3);
+
+
+    public int getAmount() {
+        return amount;
+    }
+
+    public void setAmount(int amount) {
+        this.amount = amount;
+    }
+
+    public PetrolStation(int amount) {
+        if (amount <= 0) {
+            throw new RuntimeException("PetrolStation is empty");
+        } else {
+            this.amount = amount;
+        }
+    }
+
+    public void doRefuel(int fuel) {
+        try {
+            semaphore.acquire();
+            Thread.sleep((long) (Math.random() * 7000 + 3000));
+            synchronized (this) {
+                amount -= fuel;
+            }
+        } catch (InterruptedException e) {
+            e.printStackTrace();

```

Comment:
Do 'Thread.currentThread().interrupt()` instead of printing the stack trace. If you need error message in console, use logger